enhancements/installer/kubevirt-ipi.md

```diff
@@ -0,0 +1,338 @@
+ for the machine-api to allow day 2 operations of resizing the cluster.
+
+
+## Motivation
+
+- Achieve true multi-tenancy of OpenShift were each tenant has dedicated control plane \
+ and have full control on its configuration. 
+
+### Goals
+
+- provide a way to install OpenShift on KubeVirt infrastructure using 
+  the installer - an IPI installation. (1st day)
+- implementing a cluster-api provider to provide scaling and managing the cluster
+  nodes (used by IPI, and useful for UPI, and also node management/fencing) (2nd day)
+- Provide multi-tenancy and isolation between the tenant clusters
+
+### Non-Goals
+- UPI implementation will be provided separately.
+
+## Proposal
+
+This provider enables the OpenShift Installer to provision VM resources in 
+KubeVirt infrastructure, that will be used as worker and masters of the clusters. It 
+will also create the bootstrap machine, and the configuration needed to get
+the initial cluster running by supplying a DNS service and load balancing.
+
+We want to approach deployment on OpenShift and KubeVirt as deployment on cloud similar to the
+deployments we have on public clouds as AWS and GCP rather than virtualization platform in a way 
+that the machine's network will be private, and the relevant endpoints will be exposed out of the
+cluster with platform services as we can or pods deployed in the infrastructure cluster to supply the services
+as DNS and Loadbalancing.
+
+We see two main network options for deployment over KubeVirt:
+- Deploy the tenant cluster on the pods network and use OpenShift services and routes
+to provide DNS and Load-Balancing.
+- Deploy the tenant cluster on a secondary network (using Multus) and provide DNS service and Load-Balancing
+as the same way as other KNI networking deployments using HAProxy, CoreDNS and keepalived running on the
+tenant cluster VMs. See the [baremetal ipi networking doc][baremetal-ipi-networking]
+
+ 
+
+### Implementation Details/Notes/Constraints [optional]
+
+1. Survey
+
+    The installation starts and right after the user supplies their public ssh key,\
+and then choose `KubeVirt` the installation will ask for all the relevant details\
+of the installation: **kubeconfig** for the infrastructure OpenShift, **namespace**, **storageClass**, 
+ **networkName (NAD)** and other KubeVirt specific attributes. 
+The installer will validate it can communicate with the api, otherwise it will fail to proceed.\
+
+   With that the survey continues to the general cluster name, domain name, and \
+the rest of the non-KubeVirt specific question.
+
+2. Resource creation - terraform
+
+   Terraform uses kubernetes provider to create:
+
+    - DataVolume with RHCOS image \
+         *Note* In disconnected environment the user will need to provide a local image that the installer
+          can upload to the namespace.
+    - secrets for the ignition configs of the VMs
+    - 1 bootstrap machine
+    - 3 masters
+    
+    When installing on pods network:
```

Comment:
Is it still going to be possible? Can you add an href to the relevant networking option?

Reply:
yes, when the gaps will be solved